### PR TITLE
Update getKustoDateTime with more lenient formatter

### PR DESCRIPTION
**Pull request by breriksenms**
#### Pull Request Description

The `getKustoDateTime` function uses 2 potential formatters for the LocalDateTime that either include no digits for the fraction of a second or 7 digits for the fraction of a second. We ran into a situation where the value coming back from ADX had a different number of digits and would cause this method to fail. This code updates that function to just use the built-in `ISO_LOCAL_DATE_TIME` formatter, which allows for 0-9 digits in the fraction of a second, and appends the literal 'Z' to the end of the formatter to account for that value being present in the datetimes that ADX sends back.

---

#### Future Release Comment
Updating the `getKustoDateTime` function to use the built-in `ISO_LOCAL_DATE_TIME` formatter, which allows for 0-9 digits in the fraction of a second, and append the literal 'Z' to the end of the formatter to account for that value being present in the datetimes that ADX sends back.

**Breaking Changes:**
- None

**Features:**
- None

**Fixes:**
- Use a more lenient LocalDateTime formatter for parsing in `getKustoDateTime`

### CODE REVIEW

**Comment by xatier** [comment]
Related to https://github.com/Azure/azure-kusto-java/issues/241

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 1]
```diff
@@ -465,15 +465,9 @@ public LocalDateTime getKustoDateTime(int columnIndex) {
...
-            dateTimeFormatter = new DateTimeFormatterBuilder().parseCaseInsensitive()
-                    .append(DateTimeFormatter.ofPattern(CslDateTimeFormat.KUSTO_DATETIME_PATTERN_NO_FRACTIONS)).toFormatter();
-        } else {
-            dateTimeFormatter = new DateTimeFormatterBuilder().parseCaseInsensitive()
-                    .append(DateTimeFormatter.ofPattern(CslDateTimeFormat.KUSTO_DATETIME_PATTERN)).toFormatter();
-        }
-        return LocalDateTime.parse(getString(columnIndex), dateTimeFormatter);
+        DateTimeFormatter dateTimeFormatter = new DateTimeFormatterBuilder().parseCaseInsensitive()
+                .append(DateTimeFormatter.ISO_LOCAL_DATE_TIME).appendLiteral('Z').toFormatter();
+        return LocalDateTime.parse(dateString, dateTimeFormatter);
```
Comment (ohadbitt) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:470]: You can keep the formatter as static member

**Review by ohadbitt — APPROVED** [review]
Nice!

**Comment by yogilad** [comment]
We need to add additional tests for fractions and partial values before completing this PR.
If the existing formatters are not required, they need to be deleted.
Also, there are additional formats which are (probably) used for user input which do not support timezone notation

**Comment by breriksenms** [comment]
Do you have examples of the different datetime formats that you would want this method to support?

**Comment by AsafMah** [comment]
@breriksenms This PR replaces both KUSTO_DATETIME_PATTERN_NO_FRACTIONS and KUSTO_DATETIME_PATTERN
Can you test it still works with both formats at least?

Also KUSTO_DATETIME_FORMATS in CslDateTimeFormat still doesn't seem to support timezones.

**Comment by breriksenms** [comment]
I added new tests for the getKustoDateTime function that covers the different formats it supports. I'm not certain what you want to do with the KUSTO_DATETIME_FORMATS and timezones. I think that's more for sending a datetime out to Kusto, not for receiving it. Kusto only sends back datetimes in UTC timezone. The original getKustoDateTime function only supported UTC and then only in 2 formats. This PR just expands it to support a variable number of digits in the fraction of a second.

**Pull request merged**